src/fhirtypes/common.ts

```diff
@@ -299,12 +299,13 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
       assignedReference.sdType = instanceMeta.sdType;
     }
   } else if (value instanceof FshCode) {
-    const codeSystem = tank.fish(value.system, Type.CodeSystem);
+    const [system, version] = value.system?.split('|') ?? [];
```

Comment:
I think this could lose part of the version if the version contains a `|` character. I think something like this would work:
```js
const [system, ...versionParts] = value.system?.split('|') ?? [];
const version = versionParts.join('|');
```

Reply:
This makes sense to do, but I do have one question. If I include that here, it looks like the `toFHIRCoding` method of `FshCode` (in [this line](https://github.com/FHIR/sushi/blob/master/src/fshtypes/FshCode.ts#L21)) will again lose part of the version if there is a `|` character. That would result in part of the version not included in the output.

I've updated the line, but it does pretty explicitly use `.split('|', 2)`, which will cut off anything after the first `|`. Is there any reason why we had that and I shouldn't have taken it out?

Reply:
If I wrote that code, I probably was thinking that JavaScript's `string.split` worked like Java's `[string.split](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#split(java.lang.String,int))`.  In Java, if you use a limit (e.g., `2`) then the second item in the returned array contains _the rest of the string_ even if it includes the delimiter character.  It appears JavaScript does not work that way.

Reply:
It was indeed you who wrote that code, and that is a bummer, because that's exactly what I wanted `string.split` to do, but alas, it is exactly not what Javascript's [`string.split`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split) does:

> Limit: A non-negative integer specifying a limit on the number of substrings to be included in the array. If provided, splits the string at each occurrence of the specified `separator`, but stops when `limit` entries have been placed in the array. Any leftover text is not included in the array at all.

But this at least makes me more comfortable updating that line! So I think this is set to be re-reviewed.